activestorage/lib/active_storage/service.rb

```diff
@@ -59,11 +60,19 @@ def configure(service_name, configurations)
       # Passes the configurator and all of the service's config as keyword args.
       #
       # See MirrorService for an example.
-      def build(configurator:, name:, service: nil, **service_config) # :nodoc:
+      def build(configurator:, name:, service: nil, url_config: nil, **service_config) # :nodoc:
```

Comment:
Actually, this flow isn't right. We have to convert the url to a service_config earlier in the process. It's convoluted that we're parsing in both parts. This service object shouldn't have to know anything about whether the service config came from a URL or a YAML file.